gustaf/spline/microstructure/generator.py

```diff
@@ -0,0 +1,543 @@
+          Representation of the building block defined in the unit cube
+        parametrization_function : Callable (optional)
+          Function to describe spline parameters
+        """
+        if deformation_function is not None:
+            self.deformation_function = deformation_function
+
+        if tiling is not None:
+            self.tiling = tiling
+
+        if microtile is not None:
+            self.microtile = microtile
+
+        if parametrization_function is not None:
+            self.parametrization_function = parametrization_function
+
+    @property
+    def deformation_function(self):
+        """
+        Deformation function defining the outer geometry (contour) of the
+        microstructure.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        deformation_function : spline
+        """
+        if hasattr(self, "_deformation_function"):
+            return self._deformation_function
+        else:
+            return None
+
+    @deformation_function.setter
+    def deformation_function(self, deformation_function):
+        """
+        Deformation function setter defining the outer geometry of the
+        microstructure. Must be spline type and as such inherit from
+        gustaf.GustafSpline
+
+        Parameters
+        ----------
+        deformation_function : spline
+
+        Returns
+        -------
+        None
+        """
+        if not issubclass(type(deformation_function), base.GustafSpline):
+            raise ValueError(
+                    "Deformation function must be Gustaf-Spline."
+                    " e.g. gustaf.NURBS"
+            )
+        self._deformation_function = deformation_function
+        self._sanity_check()
+
+    @property
+    def tiling(self):
+        """
+        Number of microtiles per parametric dimension
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        tiling : list<int>
+        """
+        if hasattr(self, "_tiling"):
+            return self._tiling
+        else:
+            return None
+
+    @tiling.setter
+    def tiling(self, tiling):
+        """
+        Setter for the tiling attribute, defining the number of microtiles per
+        parametric dimension
+
+        Parameters
+        ----------
+        tiling : int / list<int>
+          Number of tiles for each dimension respectively
+        Returns
+        -------
+        None
+        """
+        if not isinstance(tiling, list):
+            if not isinstance(tiling, int):
+                raise ValueError(
+                        "Tiling mus be either list of integers of integer "
+                        "value"
+                )
+        self._tiling = tiling
+        # Is defaulted to False using function arguments
+        self._sanity_check()
+        self._logd(f"Successfully set tiling to : {self.tiling}")
+
+    @property
+    def microtile(self):
```

Comment:
so do `microstructure` and `microtile` mean the same? or is `microtile` used here to go well with `tiling`? Do you plan to add another ways to produce microstructures, other than tiling?